hello_distribution/upload.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
```

Comment:
`upload.sh` can be deleted too since I haven't been including it in the other examples. There's a PR to `nerves` right now as a reminder, but I suspect that most people will be burning SD cards to try the examples rather than doing ssh updates.